main/wallet-api.md

```diff
@@ -0,0 +1,206 @@
+# Wallet API
+
+This document lists and describes the API commands available to the wallet.
+
+## Petnames
+
+The Agoric wallet uses *petnames* for issuers and purses.
+
+Petnames are your personal names for objects. No one else can see or
+modify a petname without your permission. You can think of them as
+your phone's contacts list. The actual phone number is what your phone
+uses to call someone, but for you to more easily tell who a number is
+associated with, you've assigned a petname to it, such as Mom,
+Grandpa, Kate S., etc.
+
+## Overview
+
+Wallet API commands work with the following object types:
+- `purse`: Stores assets until you withdraw them into a payment for use 
+- `issuer`: Creates empty purses and payments and maps minted assets
+  to them when assets are added or removed. Issuers verify and move
+  digital assets.
+- `offer`: Consists of what amount of what brand you're willing to
+  give, and what amount of what brand you want, as well as the
+  conditions under which the offer holder can cancel it.
+
+## Purse API Methods
+
+### `deposit()`
+- **Function:**
+  - Deposits a payment into the specified purse in the wallet. Returns the deposited amount.
+- **Required Arguments:**
+  - `pursePetname`
+    - **String**. The petname of the purse in which to deposit the specified payment.
+  - `payment`
+    - **Object**. A payment object, which is deposited into the specified purse.
+- **Optional Arguments:**
```

Comment:
Delete these sections. Just use "optional" in the description for such arguments e.g., [burn](https://agoric.com/documentation/ertp/api/issuer.html#issuer-burn-payment-optamount).

Reply:
I think it's clearer for optional and required arguments to be separate.

Reply:
That's not typical for JS apis; there's too many variants (e.g., required objects with optional subfields, arguments that default, etc.)